public/app/features/annotations/annotations_srv.ts

```diff
@@ -60,7 +60,10 @@ export class AnnotationsSrv {
         });
 
         annotations = dedupAnnotations(annotations);
-        annotations = makeRegions(annotations, options);
+        for (let i = 0; i < annotations.length; i++) {
+          const a = annotations[i];
+          a.isRegion = a.time !== a.timeEnd;
```

Comment:
I know I'm a little late to the party, but I was working on #17602, and wanted to adopt it to this new method of creating regions.
All the existing single-point annotations from Elasticsearch is rather broken right now, because the ES annotations only set `time`, not `timeEnd`, and thus they all get turned into regions.

Wouldn't it be more backwards compatible to do `a.isRegion = a.timeEnd && a.time !== a.timeEnd` or something similar.
That way the data-sources that don't provide a `timeEnd` doesn't see their annotations turned into regions?

Reply:
Think #18903 fixes this. Please merge in/rebase on master

Reply:
Ah, right - I hadn't seen that it had been fixed later - sorry. So yes, I'll rebase #17602 and re-test.